src/platforms/elastic/index.js

```diff
@@ -0,0 +1,54 @@
+import rp from 'request-promise-native'
```

Comment:
Something is making me 🤔 about this file but I can't put my finger on exactly it.  Below are a series of disorganized thoughts:

1. It seems to be a series of utility functions sort of.

2. Is this intended to be an ElasticSearch config management object?  If so we may want to actually give it a class with a descriptive name.

3. `putAllConfigs` feels like it might belong somewhere else / is doing magic (rather than having the orchestrating script handle deciding what "all configs" means and calling the putConfig) <-- should that maybe belong in the `scripts/` script instead of in here?

3b + 2. Basically right now we are half way treating this as a config management object and half way as a set of utility methods (`putConfig` and `getConfig` are both stateless, but `putAllConfigs` relies on object state).  I'm not sure we need state (and if we do, that state should be set by code outside of the object anyway through a `setConfigFiles` or something so the programmer to define that state in more interesting ways(.

It might be better to have the config file list be a parameter of `putAllConfigs` and then the `script/` calls and passes the CONFIG_FILES constant.  Just as an example of how this could be usefully more flexible, that would allow you to modify the `script` to allow the user to set the list of config files from a list of CLI passed parameters.

Feel free to ignore all of these thoughts!

Reply:
Oh, this is excellent feedback, thank you! I think I'm going to:

1. Change the `putConfig.js` script so that it _always_ assumes a list of config arguments (even if a list of just one) and iterates over `putConfig()` with them.
2. If `--all` is discovered, ignore other arguments and just iterate over `CONFIG_FILES`.\*
3. Either get rid of `putAllConfigs()` and just inline that logic in `putConfig.js`, or rewrite it to be a stateless proxy to calling `putConfig()` on a list (passed in as an arg).
4. Punt on making this a class for now. Yes, it's a series of utility functions.

\* I'm torn on whether to use `CONFIG_FILES` as a master sanitization list too, so basically if someone adds a new config file and tries to upload it, it gets rejected unless it's also been explicitly added to `CONFIG_FILES`. What do you think? If we don't use it as a master list, then `--all` would simply merge `CONFIG_FILES` with other parameters passed to the script, rather than stomp on them.

Reply:
Decided I do want `CONFIG_FILES` to be an explicit manifest of all configs available, and not allow ad hoc uploads. This is mostly to keep us honest about what `--all` does.